File: C#/Basic Auth/BasicAuth.cs
using System;
using System.IO;
using System.Net;

public class BasicAuth
{
    static public void Main()
    {
        string baseUrl = "{base url}/rest/latest/";
        string username = "API_User";
        string password = "********";

        string resource = "projects";

        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(baseUrl + resource);
        request.Credentials = new NetworkCredential(username, password);
        request.Method = WebRequestMethods.Http.Get;
        request.PreAuthenticate = true;

        HttpWebResponse response = (HttpWebResponse)request.GetResponse();

        Stream stream = response.GetResponseStream();
        StreamReader streamReader = new StreamReader(stream);
        string s = streamReader.ReadToEnd();

        Console.WriteLine(s);
    }
}

File: PHP/Basic Auth/basic_auth.php
<?PHP
$base_url = "{base url}/rest/latest/";
$username = "API_User";
$password = "********";

$resource = "projects";

$options = array(
    "http"=>array(
        "method"=>"GET",
        "header"=> "Authorization: Basic " . base64_encode("$username:$password")
    )
);
$context = stream_context_create($options);

$result = file_get_contents($base_url . $resource, false, $context);

echo $result;
?>

